controller/konnect/reconciler_generic.go

```diff
@@ -174,6 +173,13 @@ func (r *KonnectEntityReconciler[T, TEnt]) Reconcile(
 				}
 			}
 		}
+
+		if res, err := setProgrammedStatusConditionBasedOnOtherConditions(ctx, r.Client, ent); err != nil {
+			return res, err
+		} else if !res.IsZero() {
```

Comment:
Here the `Reconcile` returns `ctrl.Result{}, nil` when the returned `res`. Do you think it is simpler to directly return the return values from calling `setProgrammedStatusConditionBasedOnOtherConditions` here, like 
```
return setProgrammedStatusConditionBasedOnOtherConditions(ctx, r.Client, ent)
```
?

Reply:
👍 0fe2fbf64511d97dd9ecc96e81da7ca0d0b9784a